wrappers/src/sync_manager_cs.cpp

```diff
@@ -0,0 +1,86 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <future>
+#include <realm.hpp>
+#include <realm/util/uri.hpp>
+#include "error_handling.hpp"
+#include "marshalling.hpp"
+#include "realm_export_decls.hpp"
+#include "shared_realm_cs.hpp"
+#include "shared_realm.hpp"
+#include "sync/sync_manager.hpp"
+#include "sync/sync_user.hpp"
+#include "sync/sync_config.hpp"
+#include "sync/sync_session.hpp"
+
+using namespace realm;
+using namespace realm::binding;
+
+struct SyncConfiguration
+{
+    std::shared_ptr<SyncUser>* user;
+
+    uint16_t* url;
+    size_t url_len;
+};
+
+void (*s_refresh_access_token_callback)(std::shared_ptr<SyncUser>*, std::shared_ptr<SyncSession>*, const char* path, size_t path_len);
+
+extern "C" {
+REALM_EXPORT void realm_initialize_sync(const uint16_t* base_path_buf, size_t base_path_len, decltype(s_refresh_access_token_callback) refresh_callback)
```

Comment:
They're primitives - we can't change their contents the same way as the string buffer so there's no need to denote them as `const`.